tests/test_environments/test_gym_integration.py

```
@@ -421,6 +486,123 @@ def test_auto_port_selection(self):
+        let a single urlopen call overrun the deadline."""
+        from nemo_evaluator.environments import gym as gym_mod
+        from nemo_evaluator.environments.gym import ManagedGymEnvironment
+
+        env = ManagedGymEnvironment(server_cmd="x", port=9999, startup_timeout=0.3)
+        proc = MagicMock()
+        proc.poll.return_value = None  # still running
+        env._process = proc
+
+        timeouts: list[float] = []
+
+        def fake_urlopen(url, timeout):
```

Ruff (0.15.21)

[warning] Missing return type annotation for private function `fake_urlopen`

Add return type annotation: `NoReturn`

(ANN202)